File: 3.x.x/crates-utopia/cache/src/adapter/none.rs
use crate::adapter::Adapter;
use crate::error::CacheError;
use crate::value::{CacheValue, LoadResult, SaveResult};

/// PHP `Utopia\Cache\Adapter\None`.
#[derive(Debug, Clone, Copy, Default)]
pub struct None;

impl None {
    #[must_use]
    pub fn new() -> Self {
        Self
    }
}

impl Adapter for None {
    fn load(&self, _key: &str, _ttl: i64, _hash: &str) -> Result<LoadResult, CacheError> {
        Ok(LoadResult::Miss)
    }

    fn save(&self, _key: &str, _data: &CacheValue, _hash: &str) -> Result<SaveResult, CacheError> {
        Ok(SaveResult::Failed)
    }

    fn touch(&self, _key: &str, _hash: &str) -> Result<bool, CacheError> {
        Ok(false)
    }

    fn list(&self, _key: &str) -> Result<Vec<String>, CacheError> {
        Ok(Vec::new())
    }

    fn purge(&self, _key: &str, _hash: &str) -> Result<bool, CacheError> {
        Ok(true)
    }

    fn flush(&self) -> Result<bool, CacheError> {
        Ok(true)
    }

    fn ping(&self) -> bool {
        true
    }

    fn get_size(&self) -> Result<i64, CacheError> {
        Ok(0)
    }

    fn get_name(&self, _key: Option<&str>) -> String {
        "none".into()
    }
}

File: src/Cache/Adapter/None.php
<?php

declare(strict_types=1);

namespace Utopia\Cache\Adapter;

use Utopia\Cache\Adapter;

class None implements Adapter
{
    /**
     * @param  int  $ttl time in seconds
     * @param  string  $hash optional
     */
    public function load(string $key, int $ttl, string $hash = ''): mixed
    {
        return false;
    }

    /**
     * @param  array<int|string, mixed>|string  $data
     * @param  string  $hash optional
     * @param  int  $ttl time in seconds
     * @return bool|string|array<int|string, mixed>
     */
    public function save(string $key, array|string $data, string $hash = '', int $ttl = 0): bool|string|array
    {
        return false;
    }

    /**
     * @param  string  $hash optional
     */
    public function touch(string $key, string $hash = ''): bool
    {
        return false;
    }

    /**
     * @return string[]
     */
    public function list(string $key): array
    {
        return [];
    }

    /**
     * @param  string  $hash optional
     */
    public function purge(string $key, string $hash = ''): bool
    {
        return true;
    }

    public function flush(): bool
    {
        return true;
    }

    public function ping(): bool
    {
        return true;
    }

    public function getSize(): int
    {
        return 0;
    }

    public function getName(?string $key = null): string
    {
        return 'none';
    }
}
